src/lib/layer-manager.js

```diff
@@ -446,6 +496,86 @@ export default class LayerManager {
     }
     return error;
   }
+
+  _validateEventHandling() {
+    // Check if a deck-level mouse event has been specified
+    // and that at least one of the layers in the deck is pickable
+    if (
+      this.onLayerClick ||
+      this.onLayerHover
+    ) {
+      if (this.layers.length && !this.layers.some(layer => layer.props.pickable)) {
+        log.once(1,
+          'You have supplied a top-level input event handler (e.g. `onLayerClick`), ' +
+          'but none of your layers have set the `pickable` flag.'
+        );
+      }
+    }
+  }
+
+  /**
+   * Route click events to layers.
+   * `pickLayer` will call the `onClick` prop of any picked layer,
+   * and `onLayerClick` is called directly from here
+   * with any picking info generated by `pickLayer`.
+   * @param {Object} event  An object encapsulating an input event,
+   *                        with the following shape:
+   *                        {Object: {x, y}} offsetCenter: center of the event
+   *                        {Object} srcEvent:             native JS Event object
+   */
+  _onClick(event) {
+    const pos = event.offsetCenter;
```

Comment:
It's standard now for all events generated by `EventManager`, but you're correct that neither `offsetCenter` nor `srcEvent` are guaranteed to be present on events generated by any other event handling solution.

We're never calling `_onClick` nor `_onPointerMove` from LayerManager, those are handlers registered on the `eventManager` passed into `LayerManager.setEventParams()`. Therefore, the shim you suggest is not really applicable here.

We can do more to increase the flexibility of LayerManager's implementation of these handlers, but ultimately there is a contract between LayerManager and the passed `eventManager` that either needs to be explicit (and documented, which I did in the header comment for `setEventParams` and the `_onClick` & `_onPointerMove` handlers), or shimmed with e.g. an accessor that needs to be passed into `setEventParams()` as well.

In the latter case, the API gets more complex, and we're well into territory that IMO we don't need to venture into right now -- passing a custom `eventManager` into `LayerManager`. This feels to me like an edge case, and I think supporting it via more than clear documentation is something we should hold off on until a clear use case arises.